src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -994,4 +1056,32 @@ private static BsonValue toBsonValue(@Nullable Object queryLiteral) {
         }
         throw new FeatureNotSupportedException("Unsupported Java type: " + queryLiteral.getClass());
     }
+
+    private static final class LimitJdbcParameter extends AbstractJdbcParameter {
+
+        private final Function<Limit, Integer> parameterValueAccess;
+
+        public LimitJdbcParameter(BasicType<Integer> type, Function<Limit, Integer> parameterValueAccess) {
+            super(type);
+            this.parameterValueAccess = parameterValueAccess;
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public void bindParameterValue(
+                PreparedStatement statement,
+                int startPosition,
+                JdbcParameterBindings jdbcParamBindings,
+                ExecutionContext executionContext)
+                throws SQLException {
+            getJdbcMapping()
+                    .getJdbcValueBinder()
+                    .bind(
+                            statement,
+                            parameterValueAccess.apply(
+                                    executionContext.getQueryOptions().getLimit()),
```

Comment:
This is the magic to achieve `Limit` cache. Firstly a `JdbcParameter` was created, then a special `ParameterBinder` is created here to use the `executeContext` environment variable which shares the same `QueryOptions` so we can get the different `Limit` value at runtime to populate the static JDBC parameter placeholder.